Bugfix/install as src

DavidWLoupe opened the pull request:
## What

(Re)development of the feature install with `-src` flag. Repository will be cloned to folder outside AS and added to the AS libraries package _by reference_

## Why

With the new directory structure of lpm package repositories, it is no longer sufficient to clone the contents of the whole repository into the `logical/libraries/loupe` folder/package.

## Notes
- This PR is related to [PR20](https://github.com/loupeteam/LPM_Archive/pull/20) in the LPM_Archive repo. Most of the development occurred there and a more detailed commit history can be found in that repo.
- Co-developement occured on the ASPython library to support referenced libraries and packages. The corresponding PR is [PR2](https://github.com/loupeteam/ASPython/pull/2)

DavidWLoupe committed:
Add changes from LPM_Archive branch of same name

DavidWLoupe committed:
Update ASPython version

AndrewMusser reviewed (approved):
Looks like these are the changes we already reviewed on the archive repo, so I'm good with them.

DavidWLoupe committed:
Merge tag 'v1.0.4' into bugfix/install-as-src

DavidWLoupe committed:
Add initial pytest

DavidWLoupe committed:
Use file rather than sys.argv for finding version

DavidWLoupe committed:
Create ASProject template at beginning of testing

if necessary

DavidWLoupe committed:
Make setup/teardown class-wide

DavidWLoupe committed:
Put build test in correct location

DavidWLoupe committed:
Merge branch 'bugfix/install-as-src-pytesting' into bugfix/install-as-src

DavidWLoupe committed:
Add library attribute conversion when deploying source

DavidWLoupe committed:
Amend test to allow return of 1

1: warnings only per https://help.br-automation.com/#/en/4/programming%2Fbuild_commandline.html

DavidWLoupe committed:
Update ASPython submodule

DavidWLoupe committed:
Add building on Arm to tests

DavidWLoupe committed:
Add various dependency installation to tests

DavidWLoupe committed:
Add more tests for installing programs

DavidWLoupe committed:
Fix installation of programs

DavidWLoupe committed:
Bump version
